File: LotteryApp/AndroidLotteryApp/src/com/dyslexicApps/LotteryNumbers.java
package com.dyslexicApps;

import java.util.*;

/**
 * Created with IntelliJ IDEA.
 * User: eli
 * Date: 7/22/12
 * Time: 5:36 PM
 * To change this template use File | Settings | File Templates.
 */
public class LotteryNumbers
{
    public LotteryNumbers()
    {
        for (int i = 0; i < 6; i++)
            m_numbers.add(0);
    }

    public LotteryNumbers(String lotteryString)
    {
        ParseString(lotteryString);
    }

    private List<Integer> m_numbers = new ArrayList<Integer>();
    protected List<Integer> Numbers()
    {
         return m_numbers;
    }

    public int getNumber1() { return m_numbers.get(0); }
    public void setNumber1(int value) { m_numbers.set(0, Math.abs(value)); }

    public int getNumber2() { return m_numbers.get(1); }
    public void setNumber2(int value) { m_numbers.set(1, Math.abs(value)); }

    public int getNumber3() { return m_numbers.get(2); }
    public void setNumber3(int value) { m_numbers.set(2, Math.abs(value)); }

    public int getNumber4() { return m_numbers.get(3); }
    public void setNumber4(int value) { m_numbers.set(3, Math.abs(value)); }

    public int getNumber5() { return m_numbers.get(4); }
    public void setNumber5(int value) { m_numbers.set(4, Math.abs(value)); }

    public int getNumber6() { return m_numbers.get(5); }
    public void setNumber6(int value) { m_numbers.set(5, Math.abs(value)); }

    public String Jackpot;
    public String Status;

    public void setNumber(int index, String numberValue)
    {
        m_numbers.set(index, ConvertToInt(numberValue));
    }

    public void setNumber(int index, int numberValue)
    {
        m_numbers.set(index, numberValue);
    }

    private void ParseString(String lotteryNumbers)
    {
        String[] arr = lotteryNumbers.split("-");

        if (m_numbers.size() > 0)
            m_numbers.clear();

        for(int i = 0; i < arr.length; i++)
            m_numbers.add(ConvertToInt(arr[i]));

        Collections.sort(m_numbers);
    }

    public static LotteryNumbers ParseLotteryNumberString(String lotteryNumbers)
    {
        return new LotteryNumbers(lotteryNumbers);
    }

    public static Map<Integer, Integer> crossExamineSets(LotteryNumbers set1, LotteryNumbers set2)
    {
        Map<Integer, Integer> dict = new HashMap<Integer, Integer>();

        for (int i = 0; i < 6; i++)
        {
            for (int j = 0; j < 6; j++)
            {
                if (set1.Numbers().get(i) == set2.Numbers().get(j))
                    dict.put(i, j);
            }
        }

        return dict;
    }

    public boolean hasDuplicateValues()
    {
        List<Integer> lst = new ArrayList<Integer>();

        int temp = 0;

        for(int i = 0; i < m_numbers.size(); i++)
        {
            temp = m_numbers.get(i);

            if(temp != 0 && !lst.contains(temp))
                lst.add(temp);
        }

        //Compare the count of the list versus a distinct list (not including zero)
        return m_numbers.size() != lst.size();
    }

    public int sum()
    {
        //Not as efficient as the C# version
        int sum = 0;

        for(int i : m_numbers)
            sum += i;

        return sum;
    }

    @Override
    public String toString()
    {
        StringBuilder sb = new StringBuilder();
    
        for(int i = 0; i < m_numbers.size(); i++)
            sb.append(String.format("%02d", m_numbers.get(i))).append("-");

        sb.deleteCharAt(sb.length() - 1); //Remove trailing -
    
        return sb.toString();
    }

    private static int ConvertToInt(String strInt)
    {
        int intValue = 0;

        try
        {
            intValue = Integer.parseInt(strInt);
        }
        catch(Exception ex)
        {
            intValue = 0;
        }

        return intValue;
    }
}

File: LotteryApp/LotteryBaseLogic/LotteryNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ServerOps;

namespace LotteryBaseLogic
{
    public class LotteryNumbers
    {
        public LotteryNumbers()
        {
            for (int i = 0; i < 6; i++)
                m_numbers.Add(0);
        }

        public LotteryNumbers(string lotteryString)
        {
            ParseString(lotteryString);
        }

        #region Properties
        private List<int> m_numbers = new List<int>();
        protected List<int> Numbers
        {
            get { return m_numbers; }
        }

        public int Number1 
        { 
            get { return m_numbers[0]; }
            set { m_numbers[0] = Math.Abs(value); }
        }

        public int Number2
        {
            get { return m_numbers[1]; }
            set { m_numbers[1] = Math.Abs(value); }
        }

        public int Number3
        {
            get { return m_numbers[2]; }
            set { m_numbers[2] = Math.Abs(value); }
        }

        public int Number4
        {
            get { return m_numbers[3]; }
            set { m_numbers[3] = Math.Abs(value); }
        }

        public int Number5
        {
            get { return m_numbers[4]; }
            set { m_numbers[4] = Math.Abs(value); }
        }

        public int Number6
        {
            get { return m_numbers[5]; }
            set { m_numbers[5] = Math.Abs(value); }
        }

        public string Jackpot { get; set; }
        public string Status { get; set; }
        #endregion

        public void SetNumber(int index, string numberValue)
        {
            m_numbers[index] = numberValue.ConvertToInt32();
        }

        private void ParseString(string lotteryNumbers)
        {
            string[] arr = lotteryNumbers.Split("-".ToCharArray());

            if (m_numbers.Count > 0)
                m_numbers.Clear();

            foreach(string s in arr)
                m_numbers.Add(s.ConvertToInt32());

            m_numbers.Sort();
        }

        public static LotteryNumbers ParseLotteryNumberString(string lotteryNumbers)
        {
            return new LotteryNumbers(lotteryNumbers);
        }

        public static Dictionary<int, int> CrossExamineSets(LotteryNumbers set1, LotteryNumbers set2)
        {
            Dictionary<int, int> dict = new Dictionary<int,int>();

            for (int i = 0; i < 6; i++)
            {
                for (int j = 0; j < 6; j++)
                {
                    if (set1.Numbers[i] == set2.Numbers[j])
                        dict.Add(i, j);
                }
            }

            return dict;
        }

        public bool HasDuplicateValues()
        {
            //Compare the count of the list versus a distinct list (not including zero)
            return m_numbers.Count != m_numbers.Where(n => n != 0).Distinct().Count();
        }

        public int Sum()
        {
            return m_numbers.Sum();
        }

        public void Sort()
        {
            m_numbers.Sort();
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            foreach (int n in m_numbers)
                sb.Append(n.ToString("00")).Append("-");

            sb.Remove(sb.Length - 1, 1); //Remove trailing -

            return sb.ToString();
        }
    }
}
